integration/tests/cook/test_basic.py

```diff
@@ -1530,6 +1547,10 @@ def get_docker_info():
             util.session.delete('%s/rawscheduler?job=%s' % (self.cook_url, job_uuid))
             mesos.dump_sandbox_files(util.session, instance, job)
 
+    @unittest.skipIf(util.has_ephemeral_hosts(), 'If the cluster under test has ephemeral hosts, then it is generally '
+                                                 'a bad idea to use HOSTNAME EQUALS constraints, because it can cause '
+                                                 'the process responsible for launching hosts to launch hosts that '
+                                                 'never get used')
     def test_unscheduled_jobs(self):
         unsatisfiable_constraint = ['HOSTNAME', 'EQUALS', 'fakehost']
```

Comment:
Is it possible to use a different constraint that causes jobs to not be scheduled and avoid the issue of launching hosts that never get used?

Reply:
👍

Reply:
I changed this test to build up a queue and then assert that the "There are X jobs ahead in the queue" reason is listed